### ISSUE TITLE
All sequenced classifed to the same species

### ISSUE THREAD

**Issue by alneberg** [nf-core/ampliseq#78]
Hello again,

I managed to make my way through to the step `classifier` which just finished fine, but with a very strange result. All sequences have the exact same taxonomy:

```
[alneberg@r68 PATH]$ wc -l results/taxonomy/taxonomy.tsv
13040 results/taxonomy/taxonomy.tsv
[alneberg@r68 PATH]$ grep -c 'D_5__Nanoarchaeum;D_6__hot springs metagenome' results/taxonomy/taxonomy.tsv
13039
```

I find this strange since there are 13k unique sequences (verified) and the confidence value for the classification varies below 0.80 to 0.985 where the vast majority is > 0.90. Furthermore I blasted one of the sequences and it showed a proteobacterium as the first hit instead of the Archaea as here. 

So there has to be something wrong with the classifier I'm using. I don't know anything how to debug such a classifier though:

```
$ ls -lh work/c6/9bb8c5c1bcdb7c160e690ccccd1889/CCTACGGGNGGCWGCAG-GACTACHVGGGTATCTAATCC-99-classifier.qza
-rw-r--r-- 1 alneberg sllstore2017079 169M Mar  8 13:27 work/c6/9bb8c5c1bcdb7c160e690ccccd1889/CCTACGGGNGGCWGCAG-GACTACHVGGGTATCTAATCC-99-classifier.qza
[alneberg@rackham3 PATH]$ less work/c6/9bb8c5c1bcdb7c160e690ccccd1889/CCTACGGGNGGCWGCAG-GACTACHVGGGTATCTAATCC-99-classifier.qza
"work/c6/9bb8c5c1bcdb7c160e690ccccd1889/CCTACGGGNGGCWGCAG-GACTACHVGGGTATCTAATCC-99-classifier.qza" may be a binary file.  See it anyway?
```

It was created with the pipeline commands even though with some manual intervention as discussed previously in #77.  Any ideas? What should I do?

**Comment by d4straub** [comment]
qza files are zip archives, just use e.g. unzip. 

Since I have never come across such a problem, I had a look at the qiime2 forums and found this: https://forum.qiime2.org/t/wrong-taxonomy-qzv-file/5287/21
Looks like thats a specific problem with your V3/V4 primer set and a bug with sequence extraction from the database.
You could download and use the full length pre-trained classifier: https://data.qiime2.org/2018.6/common/silva-132-99-nb-classifier.qza

I actually do not know if the full length classifier has a significant drawback compared to a primer-specific trained classifier. Might be worth investigating.

**Comment by alneberg** [comment]
Thank you @d4straub!

I have been away completely from work the last three weeks so I am just about to catch up with this again. I think the link to the qiime forum that you posted was spot on, thanks! At the bottom of that discussion, it seems like the extracted sequence corresponding to this taxa is only 4 bases long and there are suggestions to avoid this by using a min-length parameter. Is this something that could potentially be used in the pipeline? I guess it never would make sense to have that short sequences included...

**Comment by d4straub** [comment]
You are right, these short reads are absolutely useless.
A newer QIIME2 version than the one used in this pipeline includes such a filter option: https://forum.qiime2.org/t/min-and-max-length-necessary-for-feature-classifier-extract-reads-from-reference-database/7797/2
Unfortunately I need first to finish 1.1.0 before I upgrade QIIME2 in the subsequent dev/release and include new features.
For the time being please try the full length pre-trained classifier mentioned before, it should be fine.

**Comment by alneberg** [comment]
Yes I realised that as well. In the new version of qiime, the default seem to be 50 bases, so upgrading qiime2 should hopefully fix this error right away. But you think that will have to wait until version 1.2? I will have a try with the full classifier instead now. Thanks!

**Comment by d4straub** [comment]
Should be fixed in release 1.1.2, closed for now.
